test/powershell/Modules/Microsoft.PowerShell.Utility/Write-Host.Tests.ps1

```diff
@@ -1,18 +1,91 @@
+    }
+}
+
+Describe "Write-Host test with TestHostCS" -Tags "CI" {
+
+    BeforeAll {
+        $hostmodule = Join-Path $PSScriptRoot "../../Common/TestHostCS.psm1"
+        import-module $hostmodule
+        $th = New-TestHost
+        $rs = [runspacefactory]::Createrunspace($th)
+        $rs.open()
+        $ps = [powershell]::Create()
+        $ps.Runspace = $rs
+
+        $testHostCSData = @(
+            @{ Name = 'defaults';                          Command = "Write-Host a,b,c";                                                                             returnCount = 1; returnValue = @("White:Black:a b c:NewLine") }
+            @{ Name = '-Separator';                        Command = "Write-Host a,b,c -Separator '+'";                                                              returnCount = 1; returnValue = @("White:Black:a+b+c:NewLine") }
+            @{ Name = '-Separator, colors and -NoNewLine'; Command = "Write-Host a,b,c -Separator ',' -ForegroundColor Yellow -BackgroundColor DarkBlue -NoNewline"; returnCount = 1; returnValue = @("Yellow:DarkBlue:a,b,c:NoNewLine") }
+            @{ Name = '-NoNewline:$true and colors';       Command = "Write-Host a,b -NoNewline:`$true -ForegroundColor Red -BackgroundColor Green;Write-Host a,b";  returnCount = 2; returnValue = @("Red:Green:a b:NoNewLine", "White:Black:a b:NewLine") }
+            @{ Name = '-NoNewline:$false and colors';      Command = "Write-Host a,b -NoNewline:`$false -ForegroundColor Red -BackgroundColor Green;Write-Host a,b"; returnCount = 2; returnValue = @("Red:Green:a b:NewLine","White:Black:a b:NewLine") }
+        )
+
+    }
+
+    AfterAll {
+        $rs.Close()
+        $rs.Dispose()
+        $ps.Dispose()
+    }
+
+    AfterEach {
+        $ps.Commands.Clear()
+        $th.ui.Streams.Clear()
+    }
 
-    $testData = @(
-        @{ Name = 'NoNewline';Command = "Write-Host a,b -Separator ',' -ForegroundColor Yellow -BackgroundColor DarkBlue -NoNewline"; returnValue = "a,b" }
-        @{ Name = 'Separator';Command = "Write-Host a,b,c -Separator '+'"; returnValue = "a+b+c" }
-    )
+    It "write-Host works with <Name>" -TestCases:$testHostCSData {
+        param($Command, $returnCount, $returnValue)
+        $ps.AddScript($Command).Invoke()
+        $result = $th.ui.Streams.ConsoleOutput
 
-    It "write-Host works with '<Name>' switch" -TestCases $testData -Pending:$IsOSX {
-        param($Command, $returnValue)
+        # I wonder that a console output is duplicated in Information Stream - Is it by design?
+        Write-Host $th.ui.Streams.Information.Count
+        Write-Host $th.ui.Streams.Information[0]
+        Write-Host $th.ui.Streams.Information[1]
 
-        & $powershell -noprofile $Command | Should Be $returnValue
+        $result.Count | Should Be $returnCount
+        $result[0] | Should Be $returnValue[0]
+        $result[1] | Should Be $returnValue[1]
```

Comment:
What scenarios do you have in mind? It seems these tests verify everything we need and look simple and clear.